openapi-cli/src/main/java/io/ballerina/openapi/generators/GeneratorUtils.java

```diff
@@ -461,4 +463,48 @@ public static void addImport(List<ImportDeclarationNode> imports, String module)
             imports.add(importModule);
         }
     }
+
+    /**
+     * This util class is to check the availability of the constraint support to data type.
+     */
+    public static boolean isConstraint(OpenAPI openAPI) {
+        if (openAPI.getComponents() != null && openAPI.getComponents().getSchemas() != null) {
+            Map<String, Schema> schemas = openAPI.getComponents().getSchemas();
+            for (Map.Entry<String, Schema> entry : schemas.entrySet()) {
+                Schema<?> value = entry.getValue();
+                if (isAvailableConstraint(value)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static boolean isAvailableConstraint(Schema<?> value) {
+        if (value instanceof ObjectSchema && value.getProperties() != null) {
+            for (Map.Entry<String, Schema> e : value.getProperties().entrySet()) {
```

Comment:
Cant we simplify the loop as follows?

```suggestion
            for (Schema<?> propertyValue : value.getProperties().values()) {
```

Reply:
You can further improve the logic using stream APIs (`anyMatch(Predicate<? super T> predicate)`)